NorthstarDLL/scripts/client/scriptserverbrowser.cpp

```diff
@@ -6,299 +6,105 @@
-	SQInteger serverIndex = g_pSquirrel<context>->getinteger(sqvm, 1);
+		// maxPlayerCount
+		g_pSquirrel<context>->pushinteger(sqvm, remoteServer.maxPlayers);
+		g_pSquirrel<context>->sealstructslot(sqvm, 7);
 
-	if (serverIndex >= g_pMasterServerManager->m_vRemoteServers.size())
-	{
-		g_pSquirrel<context>->raiseerror(
-			sqvm,
-			fmt::format(
-				"Tried to get playercount of server index {} when only {} servers are available",
-				serverIndex,
-				g_pMasterServerManager->m_vRemoteServers.size())
-				.c_str());
-		return SQRESULT_ERROR;
-	}
+		// requiresPassword
+		g_pSquirrel<context>->pushbool(sqvm, remoteServer.requiresPassword);
+		g_pSquirrel<context>->sealstructslot(sqvm, 8);
 
-	g_pSquirrel<context>->pushinteger(sqvm, g_pMasterServerManager->m_vRemoteServers[serverIndex].playerCount);
-	return SQRESULT_NOTNULL;
-}
+		// region
+		g_pSquirrel<context>->pushstring(sqvm, remoteServer.region, -1);
+		g_pSquirrel<context>->sealstructslot(sqvm, 9);
 
-ADD_SQFUNC("int", NSGetServerMaxPlayerCount, "int serverIndex", "", ScriptContext::UI)
-{
-	SQInteger serverIndex = g_pSquirrel<context>->getinteger(sqvm, 1);
+		// requiredMods
+		g_pSquirrel<context>->newarray(sqvm);
+		for (auto mod : remoteServer.requiredMods)
+		{
+			g_pSquirrel<context>->pushnewstructinstance(sqvm, 2);
 
-	if (serverIndex >= g_pMasterServerManager->m_vRemoteServers.size())
-	{
-		g_pSquirrel<context>->raiseerror(
-			sqvm,
-			fmt::format(
-				"Tried to get max playercount of server index {} when only {} servers are available",
-				serverIndex,
-				g_pMasterServerManager->m_vRemoteServers.size())
-				.c_str());
-		return SQRESULT_ERROR;
-	}
+			// name
+			g_pSquirrel<context>->pushstring(sqvm, mod.Name.c_str(), -1);
+			g_pSquirrel<context>->sealstructslot(sqvm, 0);
 
-	g_pSquirrel<context>->pushinteger(sqvm, g_pMasterServerManager->m_vRemoteServers[serverIndex].maxPlayers);
-	return SQRESULT_NOTNULL;
-}
+			// version
+			g_pSquirrel<context>->pushstring(sqvm, mod.Version.c_str(), -1);
+			g_pSquirrel<context>->sealstructslot(sqvm, 1);
 
-ADD_SQFUNC("string", NSGetServerID, "int serverIndex", "", ScriptContext::UI)
-{
-	SQInteger serverIndex = g_pSquirrel<context>->getinteger(sqvm, 1);
+			g_pSquirrel<context>->arrayappend(sqvm, -2);
+		}
+		g_pSquirrel<context>->sealstructslot(sqvm, 10);
 
-	if (serverIndex >= g_pMasterServerManager->m_vRemoteServers.size())
-	{
-		g_pSquirrel<context>->raiseerror(
-			sqvm,
-			fmt::format(
-				"Tried to get id of server index {} when only {} servers are available",
-				serverIndex,
-				g_pMasterServerManager->m_vRemoteServers.size())
-				.c_str());
-		return SQRESULT_ERROR;
+		g_pSquirrel<context>->arrayappend(sqvm, -2);
 	}
-
-	g_pSquirrel<context>->pushstring(sqvm, g_pMasterServerManager->m_vRemoteServers[serverIndex].id);
 	return SQRESULT_NOTNULL;
 }
 
-ADD_SQFUNC("bool", NSServerRequiresPassword, "int serverIndex", "", ScriptContext::UI)
+ADD_SQFUNC("bool", NSIsMasterServerAuthenticated, "", "", ScriptContext::UI)
```

Comment:
move this and the following functions to the top to make the diff better

Reply:
Oh I misunderstood. I moved the functions like you said to clean up the diff.